pwndbg/gdblib/vmmap.py

```diff
@@ -392,31 +392,43 @@ def coredump_maps() -> Tuple[pwndbg.lib.memory.Page, ...]:
     return tuple(pages)
 
 
-def parse_info_proc_mappings_line(line: str, parse_flags: bool) -> Optional[pwndbg.lib.memory.Page]:
+def parse_info_proc_mappings_line(
+    line: str, perms_available: bool, parse_flags: bool
+) -> Optional[pwndbg.lib.memory.Page]:
     """
     Parse a line from `info proc mappings` and return a pwndbg.lib.memory.Page
     object if the line is valid.
 
+    Example lines:
+        0x4c3000           0x4c5000     0x2000    0xc2000  rw-p   /root/hello_world/main
+        0x4c5000           0x4cb000     0x6000        0x0  rw-p
+
+    The objfile column might be empty, and the permissions column is only present in GDB versions >= 12.1
+    https://github.com/bminor/binutils-gdb/commit/29ef4c0699e1b46d41ade00ae07a54f979ea21cc
+
     Args:
         line: A line from `info proc mappings`.
 
     Returns:
         A pwndbg.lib.memory.Page object or None.
     """
-
-    # We look for lines like:
-    # ['0x555555555000', '0x555555556000', '0x1000', '0x1000', 'rw-p', '/home/user/a.out']
     try:
+        # Example line with all fields present: ['0x555555555000', '0x555555556000', '0x1000', '0x1000', 'rw-p', '/home/user/a.out']
         split_line = line.split()
 
-        # Permission info is only available in GDB versions >=12.1
-        # https://github.com/bminor/binutils-gdb/commit/29ef4c0699e1b46d41ade00ae07a54f979ea21cc
-        # Assume "rwxp" on older gdb versions
-        if len(split_line) < 6:
-            start_str, _end, size_str, offset_str, objfile = split_line
-            perm = "rwxp"
+        start_str = split_line[0]
+        _end = split_line[1]
+        size_str = split_line[2]
+        offset_str = split_line[3]
+
+        if perms_available:
+            perm = split_line[4]
+            # The objfile column may be empty.
+            objfile = split_line[5] if len(split_line) > 5 else ""
         else:
-            start_str, _end, size_str, offset_str, perm, objfile = split_line
+            perm = "rwxp"
```

Comment:
A comment on #2051 (https://github.com/pwndbg/pwndbg/pull/2051#discussion_r1510297318) suggests that we change this to "----" (or maybe "---p"). This becomes relevant because another other function in this file that parses vmmaps for core dumps first calls this function to get data from `info proc mappings`, and then enriches it with data from `maintenance info sections`. The `parse_flags` boolean was made to prevent this function from setting the perm flags in that context.